applying-stpa/l1_reqs.txt

```diff
@@ -0,0 +1,227 @@
+REQ-15: OpenAPS must inform the human actor if the insulin pump is set to use the wrong units.
+REQ-16: If the OpenAPS rig components are not connected together properly there must be an alarm, warning or error.
+REQ-17: The OpenAPS rig components must connect together securely.
+REQ-18: User manual must specify the correct hardware setup.
+REQ-19: Software must check that all required OpenAPS dependencies are present.
+REQ-20: Hardware connections need to be set up to match the configuration the human entered to the openAPS configuration.
+REQ-21: Communication between components must have an integrity check.
+REQ-22: Must have watchdog timer or heartbeat communications between systems to detect if insulin pump is working properly and warn the user if it is not.
+REQ-23: OpenAPS must alert human actor if the insulin pump has stopped giving feedback.
+REQ-24: OpenAPS shall be able to interpret error signals from insulin pump and convey to the human actor.
+REQ-25: The OpenAPS rig must have as long a battery life as possible.
+REQ-26: The OpenAPS rig must alert the user when on low power.
+REQ-27: The OpenAPS rig must have a safe exit process.
+REQ-28: Docs must have guidelines on avoiding a low power supply.
+REQ-29: If OpenAPS is having problems communicating with other devies in the loop, OpenAPS shall indicate this and fallback to the human programmed basal rates.
+REQ-30: OpenAPS wireless network signals must have a radio frequency between X MegaHertz and Y MegaHertz.
+REQ-31: OpenAPS documentation must direct the human actor to easily use the right OpenAPS software dependencies, perhaps via an automated script.
+REQ-32: If the human actor updates to incompatible software dependencies, there must be a way to restore software dependencies that work for OpenAPS easily.
+REQ-33: When backward compatibility is affected by an update, it needs to be clearly flagged.
+REQ-34: There shall be a heartbeat mechanism to monitor activeness between systems.
+REQ-35: When the OpenAPS system is running, there shall not be a period of X seconds with no active dosing recommendation from OpenAPS.
+REQ-36: OpenAPS application shall have a monitoring task that will check if tasks are taking too long to execute and warn the user if this occurs.
+REQ-37: The insulin pump must be set to the correct time.
+REQ-38: If OpenAPS is using a different time to the insulin pump, OpenAPS shall notify the human actor.
+REQ-39: OpenAPS shall be able to register human input of glucose levels.
+REQ-40: Insulin pump and OpenAPS rig must not be blocked off physically to the extent that they cannot connect, and this shall be explained in the docs.
+REQ-41: OpenAPS shall detect the local timezone and prompt the user to change the time when this changes.
+REQ-42: OpenAPS has a hardware integrity check and can communicate with all necessary peripherals.
+REQ-43: OpenAPS glucose level value must match glucose data from the CGM, unless it has been overriden by a human input.
```

Comment:
Need clarification: Does the user need to check this? Where does this requirement come from?